docs/cookbook/server-managed-meta-tags.md

```diff
@@ -0,0 +1,351 @@
+# Server Managed Meta Tags
+
+Inertia Rails can manage a page's meta tags on the server instead of on the frontend. This means that link previews (such as on Facebook, LinkedIn, etc.) will include correct meta _without server-side rendering_.
+
+Inertia Rails renders server defined meta tags into both the server rendered HTML and the client-side Inertia page props. Because the tags share unique `head-key` attributes, the client will "take over" the meta tags after the initial page load.
+
+## Setup
+
+### Server Side
+
+Simply add the `inertia_meta_tags` helper to your layout. This will render the meta tags in the `<head>` section of your HTML.
+
+```erb
+<!-- app/views/layouts/application.html.erb (or your custom layout) -->
+
+<!DOCTYPE html>
+<html>
+  <head>
+    ...
+    <%= inertia_meta_tags %> <!-- Add this inside your <head> tag -->
+  </head>
+</html>
+```
+> [!NOTE]
+> If you have a `<title>` tag in your Rails layout, make sure it has the `inertia` attribute on it so Inertia knows it should deduplicate it. The Inertia Rails generator does this for you automatically.
+
+
+### Client Side
```

Comment:
I thought about releasing this in a separate package or trying to merge this into Inertia.js core, but decided the path of least resistance was just to add instructions to copy paste the (tiny) component required to make this work.